mti_nma/steps/avgshape/avgshape.py

```diff
@@ -4,125 +4,85 @@
 from typing import Dict, List, Optional, Union
+import pandas as pd
+from aicsshparam import aicsshtools
 
 from datastep import Step, log_run_params
+from datastep.file_utils import manifest_filepaths_rel2abs
 
 from ..shparam import Shparam
+from .avgshape_utils import run_shcoeffs_analysis
 
 ###############################################################################
 
 log = logging.getLogger(__name__)
 
 ###############################################################################
 
-def run_shcoeffs_analysis(self, df):
-
-    import matplotlib.pyplot as plt
-
-    list_of_scatter_plots = [
-        ('shcoeffs_L0M0C','shcoeffs_L2M0C'),
-        ('shcoeffs_L0M0C','shcoeffs_L2M2C'),
-        ('shcoeffs_L0M0C','shcoeffs_L2M1S'),
-        ('shcoeffs_L0M0C','shcoeffs_L2M1C'),
-    ]
-
-    for id_plot, (varx,vary) in enumerate(list_of_scatter_plots):
-
-        fs = 18
-        fig, ax = plt.subplots(1,1,figsize=(6,4))
-        ax.plot(df[varx],df[vary],'o')
-        ax.set_xlabel(varx, fontsize=fs)
-        ax.set_ylabel(vary, fontsize=fs)
-        plt.tight_layout()
-        fig.savefig(
-            str(self.step_local_staging_dir / Path(f'scatter-{id_plot}.svg'))
-        )
-        plt.close(fig)
-
-    list_of_bar_plots = [
-        'shcoeffs_chi2',
-        'shcoeffs_L0M0C',
-    ]
-
-    for id_plot, var in enumerate(list_of_bar_plots):
-
-        fs = 18
-        fig, ax = plt.subplots(1,1,figsize=(12,8))
-        ax.bar(df.index, df[var])
-        ax.set_xlabel('CellId', fontsize=10)
-        ax.set_ylabel(var, fontsize=fs)
-        ax.tick_params('x', labelrotation=90)
-        plt.tight_layout()
-        fig.savefig(
-            str(self.step_local_staging_dir / Path(f'bar-{id_plot}.svg'))
-        )
-        plt.close(fig)
-
 
 class Avgshape(Step):
     def __init__(
         self,
-        direct_upstream_tasks: Optional[List["Step"]] = [Shparam],
-        config: Optional[Union[str, Path, Dict[str, str]]] = None,
+        clean_before_run=False,
+        direct_upstream_tasks: Optional[List["Step"]] = Shparam,
+        filepath_columns=["AvgShapeFilePath"],
+        config: Optional[Union[str, Path, Dict[str, str]]] = None
     ):
-        super().__init__(direct_upstream_tasks, config)
+        super().__init__(
+            clean_before_run=clean_before_run,
+            direct_upstream_tasks=direct_upstream_tasks,
+            filepath_columns=filepath_columns,
+            config=config
+        )
 
     @log_run_params
     def run(self, **kwargs):
+        """
+        This step uses the amplitudes of the spherical harmonic components
+        of the nuclear shapes in the dataset to construct an average nuclear mesh.
+        """
 
         # Get shparam manifest
-
         shparam = Shparam()
-        df = shparam.manifest.copy()
-        df = df.set_index('CellId', drop=True)
+        manifest_filepaths_rel2abs(shparam)
+        sh_df = shparam.manifest.copy()
+        sh_df = sh_df.set_index('CellId', drop=True)
```

Comment:
I'd would be consistent with variable names. For dataframes we should use always `df_[something]`. Replace this with `df_sh`.

Reply:
Great catch. I'll fix that